Stream_support/include/CGAL/IO/write_3mf.h

```diff
@@ -0,0 +1,403 @@
+  NMR::lib3mf_defaultpropertyhandler_setcolor(pDefaultPropertyHandler,
+                                              &default_color);
+
+  // release default property handler
+  NMR::lib3mf_release(pDefaultPropertyHandler);
+
+  // Set name
+  hResult = NMR::lib3mf_object_setnameutf8(*pMeshObject, name.c_str());
+  if (hResult != LIB3MF_OK) {
+    std::cerr << "could not set object name: " << std::hex << hResult << std::endl;
+    NMR::lib3mf_getlasterror(*pMeshObject, &nErrorMessage, &pszErrorMessage);
+    std::cerr << "error #" << std::hex << nErrorMessage << ": " << pszErrorMessage << std::endl;
+    NMR::lib3mf_release(*pMeshObject);
+    NMR::lib3mf_release(pModel);
+    return false;
+  }
+  return add_build_item(pModel, *pMeshObject);
+}
+
+template<typename PointRange, typename Color>
+bool write_point_cloud_to_model(const PointRange& points,
+                                const Color& color,
+                                const std::string& name,
+                                NMR::PLib3MFModelMeshObject** pMeshObject,
+                                NMR::PLib3MFModel * pModel
+                                )
+{
+  std::string pc_name = name;
+  pc_name.append("_cgal_pc");
+  return write_points(points, color, pc_name, pMeshObject, pModel);
+}
+
+template<typename PointRange, typename Color>
+bool write_polyline_to_model(const PointRange& points,
+                             const Color& color,
+                             const std::string& name,
+                             NMR::PLib3MFModelMeshObject** pMeshObject,
+                             NMR::PLib3MFModel * pModel
+                             )
+{
+  std::string pc_name = name;
+  pc_name.append("_cgal_pl");
+  return write_points(points, color, pc_name, pMeshObject, pModel);
+}
+
+/*!
+ * \brief write_soups_to_3mf will write the polygon soups contained in all_points and
+ *  all_polygons into the file named `file_name`, in the 3mf format.
+ * \tparam PointRanges a model of the concepts `RandomAccessContainer` and
+ *  `BackInsertionSequence` whose `value type` is
+ * a model of the concepts `RandomAccessContainer` and `BackInsertionSequence`
+ *  whose `value type` is the point type.
+ * \tparam PolygonRanges a model of the concept `RandomAccessContainer` whose
+ *  `value_type` is a model of the concept `RandomAccessContainer`
+ * whose `value_type` is a model of the concept `RandomAccessContainer` whose
+ *  `value_type` is std::size_t.
+ * \param file_name the name of the 3mf file to write.
+ * \param all_points a `PointRanges` that contains the points of the soups
+ *  to write.
+ * \param all_polygons a `PolygonRanges` that contains the triangles of the
+ *  soups in `file_name`.
+ * \param names will contains the name of each mesh in `file_name`.
+ * \return `true` if the writing is successful, `false` otherwise.
+ */
+template<typename PointRanges, typename PolygonRanges>
+bool write_soups_to_3mf(const std::string& file_name, 
```

Comment:
`write_triangle_soups_to_3mf`?

Reply:
Same remark as above, shall we have a function for model of `FaceListGraph` and  `HalfedgeListGraph`?

Reply:
That could be less painful to use indeed.